nautobot_device_lifecycle_mgmt/jobs/lifecycle_reporting.py

```diff
@@ -16,6 +19,70 @@
 name = "Device/Software Lifecycle Reporting"  # pylint: disable=invalid-name
 
 
+class DeviceHardwareNoticeFullReport(Job):
+    """Checks if devices are linked to hardware notices."""
+
+    name = "Device Hardware Notice Report"
+    description = "Creates data for reporting on hardware notices."
+    read_only = False
+
+    class Meta:
+        """Meta class for the job."""
+
+        has_sensitive_variables = False
+
+    def run(self) -> None:  # pylint: disable=arguments-differ
+        """Check if device is affected by a hardware notice."""
+        job_run_time = datetime.now()
+        notice_count = 0
+        device_count = 0
+
+        # Process devices through HardwareLCM.device_type
+        for notice in HardwareLCM.objects.all():
+            if notice.device_type:
+                devices_qs = Device.objects.filter(device_type=notice.device_type)
+                for device in devices_qs:
+                    if notice.end_of_support:
+                        is_supported = datetime.today().date() <= notice.end_of_support
+                    else:
+                        is_supported = True
+                    try:
+                        hardware_notice_result, _ = DeviceHardwareNoticeResult.objects.get_or_create(device=device)
+                        hardware_notice_result.hardware_notice = notice
+                        hardware_notice_result.is_supported = is_supported
+                        hardware_notice_result.last_run = job_run_time
+                        hardware_notice_result.run_type = choices.ReportRunTypeChoices.REPORT_FULL_RUN
+                        hardware_notice_result.validated_save()
+                        device_count += 1
+                    except Exception as err:
+                        self.logger.error(f"Error creating hadware notice result {err}")
+                notice_count += 1
+        # Process all devices skipping devices already processed in the previous step
+        for device in Device.objects.all():
+            try:
+                DeviceHardwareNoticeResult.objects.get(
+                    device=device,
+                    last_run=job_run_time,
+                    run_type=choices.ReportRunTypeChoices.REPORT_FULL_RUN,
+                )
+                continue
+            except ObjectDoesNotExist:
+                try:
+                    hardware_notice_result, _ = DeviceHardwareNoticeResult.objects.get_or_create(device=device)
+                    hardware_notice_result.hardware_notice = None
+                    hardware_notice_result.is_supported = True
```

Comment:
When a device is first released it is supported and will remain supported until the date on the hardware notice passes. A hardware notice will not exist for devices that are not at or near end of life, so support is assumed if a hardware notice does not exist. If we were to require hardware notices to exist for a device to be "supported" then users would be forced to create fake hardware notices with made up future eox dates on them.

Reply:
You make valid points. It might be worthwile to ask a wider audience to make sure we get this right. What I think we can already do however is to add a toggle in filters, and a column in reporting, that allows end users to filter results based on whether a matching notice exists. It would be something like what we've done for the software reporting:

![image](https://github.com/user-attachments/assets/11fd83a2-5560-4d5c-b558-9a5ca087a93d)

Reply:
Yes, I would like to get some more opinions on this. I think the pool of devices that both have a hardware notice available and are still supported would be quite small considering that EoX data isn't created until an announcement for a product's imminent end of life is made.